server/src/http/feature_refresher.rs

```diff
@@ -0,0 +1,467 @@
+                                api_key: refresh.token.token.clone(),
+                                etag: refresh.etag
+                            }).await;
+
+                            match features_result {
+                                Ok(feature_response)  => match feature_response {
+                                    ClientFeaturesResponse::NoUpdate(_) => {
+                                        debug!("No update needed. Will update last check time");
+                                        self.update_last_check(&refresh.token.clone());
+                                    }
+                                    ClientFeaturesResponse::Updated(features, etag) => {
+                                        debug!("Got updated client features. Updating features");
+                                        let key = cache_key(refresh.token.clone());
+                                        self.update_last_refresh(&refresh.token, etag);
+                                        self.features_cache.entry(key.clone()).and_modify(|existing_data| {
+                                            *existing_data = existing_data.clone().upsert(features.clone());
+                                        }).or_insert(features.clone());
+                                        if self.engine_cache.contains_key(&key) {
+                                            self.engine_cache.entry(key.clone()).and_modify(|engine| {
+                                                if let Some(f) = self.features_cache.get(&key) {
+                                                        let mut new_state = EngineState::default();
+                                                        new_state.take_state(f.clone());
+                                                        *engine = new_state;
+                                                }
+                                            });
+                                        } else {
+                                            let mut new_state = EngineState::default();
+                                            new_state.take_state(features.clone());
+                                            self.engine_cache.insert(key, new_state);
+                                        }
+                                        if let Some(persist) = self.persistence.clone() {
+                                            let feature_save = persist.save_features(
+                                                self.features_cache
+                                                    .iter()
+                                                    .map(|e| (e.key().clone(), e.value().clone()))
+                                                    .collect());
+
+                                            let refresh_targets_save = persist.save_refresh_targets(self.tokens_to_refresh.iter().map(|e| e.value().clone()).collect());
```

Comment:
Same as https://github.com/Unleash/unleash-edge/pull/89#discussion_r1126482403 for these 2 saves to persistent store. Ideally it would be nicer to only handle the in-memory cache / state layer without having to worry about read/save to store.

Reply:
Addressed in https://github.com/Unleash/unleash-edge/pull/89#discussion_r1126464090